fontforge/splinestroke.c

```diff
@@ -708,7 +708,7 @@ return;		/* Essentially colinear */ /* Won't be perfect because control points l
 	    p->needs_point_left = p->needs_point_right = false;
 	    p->left_hidden = bends_left;
 	    p->right_hidden = !bends_left;
-	    if ( rot.x<=fabs(diff_angle.x) ) {
+	    if ( rot.x<=diff_angle.x || (diff_angle.x <= -1 && rot.x <= -0.999999) ) { /* close enough */
```

Comment:
But what I do find strange this same code [doesn't cause an infinite loop](https://github.com/fontforge/fontforge/issues/2025#issuecomment-231816669) in the 2012 version.